src/java.base/share/classes/java/nio/Buffer.java

```diff
@@ -821,6 +822,18 @@ public MemorySegmentProxy bufferSegment(Buffer buffer) {
                     return buffer.segment;
                 }
 
+                @Override
+                public Scope.Handle acquireScope(Buffer buffer, boolean async) {
+                    var scope = buffer.scope();
+                    if (scope == null || scope.isImplicit()) {
```

Comment:
The null check is still needed - for "regular" buffers (that do not have a scope).

The `isImplicit` check can probably be removed. It is an unproven mirco-optimization to avoid the bloat of chaining unnecessary handles. It could be removed, and dropped altogether or pushed into the IOUtil code that does the chaining ( do not create a new Releaser for a handle that we've seen before, I assume handles can be compared by identity? )  Maybe I'll just drop this, and revisit later when benchmarking.

Reply:
I see - I wonder if it would make sense to use global scope in these cases. But yes, this is probably best tackled in a separate patch.